test-framework/remote/src/main/java/org/keycloak/test/framework/remote/timeoffset/TimeOffsetSupplier.java

```diff
@@ -0,0 +1,65 @@
+package org.keycloak.test.framework.remote.timeoffset;
+
+import org.apache.http.client.HttpClient;
+import org.keycloak.test.framework.injection.InstanceContext;
+import org.keycloak.test.framework.injection.LifeCycle;
+import org.keycloak.test.framework.injection.RequestedInstance;
+import org.keycloak.test.framework.injection.Supplier;
+import org.keycloak.test.framework.injection.SupplierOrder;
+import org.keycloak.test.framework.remote.RemoteProviders;
+import org.keycloak.test.framework.server.KeycloakServerConfigBuilder;
+import org.keycloak.test.framework.server.KeycloakServerConfigInterceptor;
+import org.keycloak.test.framework.server.KeycloakUrls;
+
+public class TimeOffsetSupplier implements Supplier<TimeOffSet, InjectTimeOffSet>, KeycloakServerConfigInterceptor<TimeOffSet, InjectTimeOffSet> {
+    @Override
+    public Class<InjectTimeOffSet> getAnnotationClass() {
+        return InjectTimeOffSet.class;
+    }
+
+    @Override
+    public Class<TimeOffSet> getValueType() {
+        return TimeOffSet.class;
+    }
+
+    @Override
+    public TimeOffSet getValue(InstanceContext<TimeOffSet, InjectTimeOffSet> instanceContext) {
+        var httpClient = instanceContext.getDependency(HttpClient.class);
+        var remoteProviders = instanceContext.getDependency(RemoteProviders.class);
+        KeycloakUrls keycloakUrls = instanceContext.getDependency(KeycloakUrls.class);
+
+        int initOffset = instanceContext.getAnnotation().offset();
+        return new TimeOffSet(httpClient, keycloakUrls.getMasterRealm(), initOffset);
+    }
+
+    @Override
+    public boolean compatible(InstanceContext<TimeOffSet, InjectTimeOffSet> a, RequestedInstance<TimeOffSet, InjectTimeOffSet> b) {
+        return true;
+    }
+
+    @Override
+    public LifeCycle getDefaultLifecycle() {
+        return LifeCycle.METHOD;
+    }
+
+    @Override
+    public void close(InstanceContext<TimeOffSet, InjectTimeOffSet> instanceContext) {
+        if (instanceContext.getLifeCycle() != LifeCycle.METHOD) {
+            TimeOffSet timeOffSet = instanceContext.getValue();
+            if (timeOffSet.hasChanged()) {
+                timeOffSet.set(0);
+            }
+        }
+    }
+
+    @Override
+    public int order() {
+        return SupplierOrder.BEFORE_KEYCLOAK_SERVER;
+        // Implementing the KeycloakServerConfigInterceptor is a workaround for RemoteProvidersSupplier to work
+    }
+
+    @Override
+    public KeycloakServerConfigBuilder intercept(KeycloakServerConfigBuilder serverConfig, InstanceContext<TimeOffSet, InjectTimeOffSet> instanceContext) {
+        return serverConfig;
+    }
```

Comment:
Strange that this is needed, or that it works; might need to tweak this a little bit so we can allow dependencies to be server config interceptors as well. We might need to add something like getDependencies or something to suppliers :/